src/server/game/AI/SmartScripts/SmartScript.cpp

```diff
@@ -145,7 +145,16 @@ void SmartScript::ProcessEventsFor(SMART_EVENT e, Unit* unit, uint32 var0, uint3
             ConditionSourceInfo info = ConditionSourceInfo(unit, GetBaseObject(), me ? me->GetVictim() : nullptr);
 
             if (sConditionMgr->IsObjectMeetToConditions(info, conds))
-                ProcessEvent(*i, unit, var0, var1, bvar, spell, gob);
+            {
+                ASSERT(executionStack.empty());
```

Comment:
I am always a bit unsure about `ASSERTs` instead of a LOG_ERROR(), because these make crashes the server

Reply:
In case `executionStack.empty()` is not true it means something messed with the stack and the behavior is going to be different than the original. 

This shouldn't happen and if somehow is it's a flaw that should be fixed. I believe logging it would suggest the intention that it's expected and will be handled, which is not the case.